extension/src/router_channel.rs

```diff
@@ -154,7 +151,7 @@ impl RouterChannel {
         self.lambda_id,
         self.channel_id
       );
-      while futures::future::poll_fn(|ctx| self.sender.poll_ready(ctx))
+      if futures::future::poll_fn(|ctx| self.sender.poll_ready(ctx))
```

Comment:
I'm not sure if this is a logic change?

Reply:
It can be changed to an `if` because it can never loop due to the final statement of the block being a `panic`. See: https://rust-lang.github.io/rust-clippy/master/index.html#/never_loop